src/Data/Array.purs

```diff
@@ -987,15 +987,15 @@ groupBy op xs =
     STA.unsafeFreeze result
 
 -- | Group equal elements of an array into arrays, using the specified
--- | equivalence relation to determine equality.
+-- | comparison relation to determine equality.
 -- |
```

Comment:
How should docs be updated to better communicate what this does?

Reply:
I think we could replace “comparison relation” by “partial ordering“, as in the documentation of sortBy, here.

Reply:
Those docs are actually incorrect - it's expecting a total ordering, not a partial one. I think they must be an oversight left over from a short period in the distant past where we said Ord represented a partial ordering rather than a total ordering. We call this a "comparison function" in Data.Array.ST, and we say "where elements are compared according to the specified ordering" in Data.List. I think either of those are fine.

Reply:
I've reverted back to using `comparison function`